test/api.mocha.coffee

```diff
@@ -257,15 +265,129 @@ describe "API", ->
     describe "Groups", ->
       group = undefined
       before (done) ->
-        request.post(baseURL + "/groups").send(
-          name: "TestGroup"
-          type: "party"
-        ).end (res) ->
-          expectCode res, 200
-          group = res.body
-          expect(group.members.length).to.be 1
-          expect(group.leader).to.be user._id
-          done()
+       request.post(baseURL + "/groups").send(
+         name: "TestGroup"
+         type: "party"
+       ).end (res) ->
+         expectCode res, 200
+         group = res.body
+         expect(group.members.length).to.be 1
+         expect(group.leader).to.be user._id
+         done()
+
+      describe "Guilds", ->
+
+        before (done) ->
+          User.findByIdAndUpdate user._id,
+            $set:
+              "balance": 4
+            , (err, _user) ->
+              done()
+
+        describe "Private Guilds", ->
+          guild = undefined
+          before (done) ->
+            request.post(baseURL + "/groups").send(
+              name: "TestPrivateGroup"
+              type: "guild"
+              privacy: "private"
+            ).end (res) ->
+              expectCode res, 200
+              guild = res.body
+              expect(guild.members.length).to.be 1
+              expect(guild.leader).to.be user._id
+              #Add members to guild
+              async.waterfall [
+                (cb) ->
+                  registerManyUsers 15, cb
+
+                (_members, cb) ->
+                  members = _members
+
+                  joinGuild = (member, callback) ->
+                    request.post(baseURL + "/groups/" + guild._id + "/join")
+                      .set("X-API-User", member._id)
+                      .set("X-API-Key", member.apiToken)
+                      .end ->
+                        callback(null, null)
+
+                  async.map members, joinGuild, (err, results) -> cb()
+
+              ], done
+
+          it "includes user in private group member list when user is a member", (done) ->
+            #Verfiy that when a user query's for a group they are in the group if they are a member
```

Comment:
About to merge this in, so don't make any changes, but a few notes.

I don't like a lot of comments in `it` blocks, unless the code is really very difficult to understand. And if that code is really difficult to understand, maybe the test needs to be re-written.

In most cases, the string after the `it` should be able to describe adequately what is being tested. I don't see much value in these comments, so I'm deleting them. The comments in the before block, where it's telling you what the code is doing when it's a little obscure (ie, async running to create new members) are great.

`.send()` is not required on the `request` object. I've removed the empty ones.